packages/Radio/src/Radio.js

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import PropTypes from "prop-types";
+import nanoid from "nanoid";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import CheckIcon from "@paprika/icon/lib/Check";
+import radioStyles from "./Radio.styles";
+import Group from "./components/Group";
+
+const propTypes = {
+  a11yText: PropTypes.string,
+  children: PropTypes.node,
+  isChecked: PropTypes.bool,
+  isDisabled: PropTypes.bool,
+  canDeselect: PropTypes.bool,
+  onClick: PropTypes.func.isRequired,
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  a11yText: null,
+  isChecked: false,
+  canDeselect: false,
+  children: null,
+  isDisabled: false,
+  size: ShirtSizes.MEDIUM,
+};
+
+function Radio(props) {
+  const { a11yText, children, isChecked, canDeselect, isDisabled, size, onClick, ...moreProps } = props;
+  const radioId = React.useRef(nanoid()).current;
+  const inputRef = React.useRef(null);
+
+  const handleKeyDown = event => {
+    if (
+      // Prevent scrolling the page with a spacerbar keypress
+      event.key === " " ||
+      // Prevent submitting forms in IE/Edge with and enter keypress
+      event.key === "Enter"
+    ) {
+      event.preventDefault();
+    }
+  };
+
+  const handleKeyUp = event => {
+    const isTriggerKey = event.key === " "; // space key
+    if (!isDisabled && isTriggerKey) {
+      onClick(event);
+    }
+  };
+
+  const styleProps = {
+    hasLabel: !!children,
+    size,
+  };
+
+  const inputProps = {
+    readOnly: true,
+    onClick,
+    checked: isChecked,
+    disabled: isDisabled,
+    id: radioId,
+    name: children,
```

Comment:
This isn't quite right.  All the Radios in this group need to have the same name.  It should be a prop on `<Radio.Group>`, but if it's not provided it can be a `nanoid` generated by the Group component and provided to each Radio.